### PR TITLE
add example YAMLs for deploying external-resizer

### PR DESCRIPTION
**Pull request by mlmhl**
ref: https://github.com/kubernetes-csi/external-resizer/issues/13

### CODE REVIEW

File: deploy/kubernetes/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            - name: socket-dir
+              mountPath: /var/lib/csi/sockets/pluginproxy/
+
+        - name: mock-driver
+          image: quay.io/k8scsi/mock-plugin
+          imagePullPolicy: "IfNotPresent"
+          env:
+            - name: CSI_ENDPOINT
+              value: /var/lib/csi/sockets/pluginproxy/mock.socket
+          volumeMounts:
```
Comment (gnufied) [inline deploy/kubernetes/deployment.yaml]: Should we be consistent with other external components that mostly use "/csi/csi.sock" ?

Comment (mlmhl) [reply deploy/kubernetes/deployment.yaml]: Currently only external snapshotter uses `/csi/csi.sock`: https://github.com/kubernetes-csi/external-snapshotter/blob/f59b58d16488bbb9c15a263bf1b2685c3c4c4441/deploy/kubernetes/setup-csi-snapshotter.yaml#L94, both provisioner and attacher use `/var/lib/csi/sockets/pluginproxy/mock.socket`: https://github.com/kubernetes-csi/external-provisioner/blob/f588078fd1f33dd7e9557cceba4c05e3562683fd/deploy/kubernetes/statefulset.yaml#L41, https://github.com/kubernetes-csi/external-attacher/blob/0e6f5f22774f89d2d2c396896c462a4fb3587902/deploy/kubernetes/deployment.yaml#L53.
So I tend to be consistent with the provisioner and attacher as they all use `mock driver`, while snapshotter uses `hostpath`.

**Pull request merged**